core/kogitoservice/deployer.go

```diff
@@ -89,6 +90,10 @@ func NewServiceDeployer(context *operator.Context, definition ServiceDefinition,
 	if len(definition.Request.NamespacedName.Namespace) == 0 && len(definition.Request.NamespacedName.Name) == 0 {
 		panic("No Request provided for the Service Deployer")
 	}
+	if infraHandler == nil {
```

Comment:
Many places in the code have calls to `infraHandler`. Actually, many unit tests were panicking with NPEs but were passing despite that. 

This way, we only log that the feature is not enabled and won't panic the branch.

Reply:
> InfraHandler should pass to nil in product operator only because product operator currently don't support that feature

It's passing nil and in our branches we are using the reference without checking, that's why it's a potential bug.

> Apart from it, In community operator if infraHandler is not passed in Unit test cases then we need to create/pass it. But it should never be nil. 

If not passed the reference is nil, then panic.

> But it should never be nil. 

We can't guarantee that with the old code. It's nil if you just created.

> I don't think add NoOps handler is right approach here. 

Ok, what is your recommendation then?